### PR TITLE
fix assertion error due to race condition in stop method

### PR DESCRIPTION
**Pull request by fredo**
I encountered a flaky test which hit an assertion error in `process_call_messages` where the `web_rtc_manager` was None. 

This is possible though during the call of the transport's `stop()` method which is context switching after the web_rtc_manager was set back to None. Due to the context switch incoming messages can still be processed after the the web rtc manager was set back to None. 

The fix changes the assert to a check agains the stop event, which should always be used to indicate that the transport is not running.

https://app.circleci.com/pipelines/github/raiden-network/raiden/12173/workflows/dee5fe1d-d0d4-4875-aca9-5b2316a5a7e2/jobs/203912

### CODE REVIEW

File: raiden/network/transport/matrix/transport.py [hunk 1]
```diff
@@ -941,7 +941,9 @@ def _process_call_messages(self, call_messages: List[ReceivedCallMessage]) -> No
             call_messages: List of signalling messages
         """
 
-        assert self._web_rtc_manager is not None, "must be set"
+        if self._stop_event.is_set():
+            return
+
```
Comment (istankovic) [inline raiden/network/transport/matrix/transport.py:946]: Hm, why is this necessary? Isn't `_handle_messages` going to return early in this case?
Also, why did you remove the assert?

Comment (fredo) [reply raiden/network/transport/matrix/transport.py:946]: the assert needs to stay. I explained in the PR. This assertion error was hit in a test

Comment (fredo) [reply raiden/network/transport/matrix/transport.py:946]: but actually good point how this is possible 

I added the CI failure above

**Comment by fredo** [comment]
I found the potential case. After `handle_messages` is called the method `_validate_matrix_messages` can actually context switch. If the displayname is missing, the displaynameCache will fetch it from the Matrix server causing a context switch. If after that the stop() method is called and it then resumes to the greenlet, the assertion error will be hit in `process_call_messages`

**Review by istankovic — APPROVED** [review]
Looks ok (it'd be nice to squash the two commits though).
